config/thisroot.sh

```diff
@@ -24,55 +24,56 @@ drop_from_path()
                           -e "s;^${drop}\$;;g"`
 }
 
-if [ -n "${ROOTSYS}" ] ; then
-   old_rootsys=${ROOTSYS}
-fi
-
-SOURCE=${BASH_ARGV[0]}
-if [ "x$SOURCE" = "x" ]; then
-    SOURCE=${(%):-%N} # for zsh
-fi
-
-if [ "x${SOURCE}" = "x" ]; then
-    if [ -f bin/thisroot.sh ]; then
-        ROOTSYS="$PWD"; export ROOTSYS
-    elif [ -f ./thisroot.sh ]; then
-        ROOTSYS=$(cd ..  > /dev/null; pwd); export ROOTSYS
-    else
-        echo ERROR: must "cd where/root/is" before calling ". bin/thisroot.sh" for this version of bash!
-        ROOTSYS=; export ROOTSYS
-        return 1
-    fi
-else
-    # get param to "."
-    thisroot=$(dirname ${SOURCE})
-    ROOTSYS=$(cd ${thisroot}/.. > /dev/null;pwd); export ROOTSYS
-fi
-
-if [ -n "${old_rootsys}" ] ; then
+drop_from_all_libs()
+{
```

Comment:
put the if condition inside the clean_environment function, which encapsulates everything except the manpath setting, then call it directly from main